### ISSUE TITLE
no-unused-props: false positives — many real props usage patterns are not tracked

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#532]
### Describe the bug

`no-unused-props` reports properties as unused even though they are actually consumed, for many common CDK coding patterns. The usage tracking only recognizes a narrow set of AST shapes:

- whole-props usage is only detected when the `props` identifier is a direct argument of a `CallExpression`, a `ReturnStatement` argument, an `ArrayExpression` element, or an object `Property` value (`DirectPropsUsageVisitor.visitIdentifier`)
- destructuring only collects `Property` keys (`findPropertyNames`), ignoring `RestElement`
- member access tracking requires a non-computed `Identifier` property and an instance variable stored via a plain `Identifier` `this.x = props` at the top level of the constructor
- method-call analysis only follows `this.method(props)` calls made directly in the constructor body

All of the following patterns are false positives (verified on both plugins; each snippet is a complete repro):

**1. props passed as a whole to a `NewExpression`**

```ts
import { Queue } from "aws-cdk-lib/aws-sqs";
import { Construct } from "constructs";

interface NewExprProps {
  readonly fifo?: boolean;
}

export class NewExprConstruct extends Construct {
  constructor(scope: Construct, id: string, props: NewExprProps) {
    super(scope, id);
    new Queue(this, "Queue", props);
  }
}
```

**2. spread**

```ts
interface SpreadProps {
  readonly versioned?: boolean;
}

export class SpreadConstruct extends Construct {
  constructor(scope: Construct, id: string, props: SpreadProps) {
    super(scope, id);
    new Bucket(this, "Dest", { ...props, enforceSSL: true });
  }
}
```

**3. alias consumed as a whole**

```ts
interface AliasProps {
  readonly versioned?: boolean;
}

export class AliasConstruct extends Construct {
  constructor(scope: Construct, id: string, props: AliasProps) {
    super(scope, id);
    const bucketProps = props;
    new Bucket(this, "Dest", bucketProps);
  }
}
```

**4. `in` operator**

```ts
interface InOpProps {
  readonly encryption?: boolean;
}

export class InOpConstruct extends Construct {
  constructor(scope: Construct, id: string, props: InOpProps) {
    super(scope, id);
    if ("encryption" in props) {
      console.log("encryption is set");
    }
  }
}
```

**5. rest destructuring**

```ts
interface RestProps {
  readonly logPrefix: string;
  readonly versioned?: boolean;
}

export class RestConstruct extends Construct {
  constructor(scope: Construct, id: string, props: RestProps) {
    super(scope, id);
    const { logPrefix, ...bucketProps } = props;
    console.log(logPrefix);
    new Bucket(this, "Dest", bucketProps);
  }
}
```

**6. computed access with a string literal**

```ts
interface ComputedProps {
  readonly bucketName: string;
}

export class ComputedConstruct extends Construct {
  constructor(scope: Construct, id: string, props: ComputedProps) {
    super(scope, id);
    console.log(props["bucketName"]);
  }
}
```

**7. props stored in an ECMAScript `#private` field**

```ts
interface PrivateFieldProps {
  readonly bucketName: string;
}

export class PrivateFieldConstruct extends Construct {
  readonly #props: PrivateFieldProps;

  constructor(scope: Construct, id: string, props: PrivateFieldProps) {
    super(scope, id);
    this.#props = props;
  }

  public bucketName(): string {
    return this.#props.bucketName;
  }
}
```

**8. transitive `this.method(props)` chains**

```ts
interface ChainProps {
  readonly bucketName: string;
}

export class ChainConstruct extends Construct {
  constructor(scope: Construct, id: string, props: ChainProps) {
    super(scope, id);
    this.setup(props);
  }

  private setup(props: ChainProps): void {
    this.createBucket(props); // calls made inside called methods are not followed
  }

  private createBucket(props: ChainProps): void {
    new Bucket(this, "Dest", { bucketName: props.bucketName });
  }
}
```

### Expected Behavior

No `unusedProp` errors for any of the snippets above — every property is (potentially) consumed. Patterns 1, 2, 3 in particular (`new Bucket(this, id, props)`, `{ ...props, override }`, passing a props alias down) are idiomatic CDK code.

### Current Behavior

Every snippet reports its properties as unused. ESLint output:

```
repro/no-unused-props/new-expression.ts    10:45  error  Property 'fifo' is defined in props but never used        awscdk/no-unused-props
repro/no-unused-props/spread.ts            10:45  error  Property 'versioned' is defined in props but never used   awscdk/no-unused-props
repro/no-unused-props/alias-whole-use.ts   10:45  error  Property 'versioned' is defined in props but never used   awscdk/no-unused-props
repro/no-unused-props/in-operator.ts        9:45  error  Property 'encryption' is defined in props but never used  awscdk/no-unused-props
repro/no-unused-props/rest-destructuring.ts 11:45 error  Property 'versioned' is defined in props but never used   awscdk/no-unused-props
repro/no-unused-props/computed-access.ts    9:45  error  Property 'bucketName' is defined in props but never used  awscdk/no-unused-props
repro/no-unused-props/private-field.ts     11:45  error  Property 'bucketName' is defined in props but never used  awscdk/no-unused-props
repro/no-unused-props/method-chain.ts      10:45  error  Property 'bucketName' is defined in props but never used  awscdk/no-unused-props
```

Oxlint reports the identical set (10 problems including a control case, same lines/properties).

### Plugin

eslint-plugin-awscdk, oxlint-plugin-awscdk

### Plugin Version

eslint-plugin-awscdk 4.4.0, oxlint-plugin-awscdk 1.0.0

### Language Version

TypeScript 6.0.3 (eslint-plugin-awscdk) / TypeScript 7 preview via oxlint-tsgolint (oxlint-plugin-awscdk)

### Additional Context

- Verified on `main` (35b889a); `packages/eslint/src` is unchanged since the 4.4.0 tag and `packages/oxlint/src` since the 1.0.0 tag, so the released versions are affected.
- Environment: macOS, Node.js 24.18.0, eslint 10.8.1, typescript-eslint 8.67.0, oxlint 1.79, rule isolated with `"awscdk/no-unused-props": "error"`.
- Root causes (same code in both packages):
  - patterns 1-4: `packages/*/src/rules/no-unused-props/visitor/direct-props-usage-visitor.ts` — `visitIdentifier` switch handles only `CallExpression` / `ReturnStatement` / `ArrayExpression` / `Property` parents; `packages/*/src/rules/no-unused-props/visitor/props-alias-visitor.ts` only tracks `alias.prop` member access
  - pattern 5: `packages/*/src/core/ast-node/finder/property-name.ts` ignores `RestElement`
  - pattern 6: `packages/*/src/rules/no-unused-props/props-usage-tracker.ts` — `markAsUsedForMemberExpression` requires `property.type === Identifier` and ignores `computed` (note: `props[key]` with a variable also wrongly marks the *variable name* as a used prop)
  - pattern 7: `props-usage-analyzer.ts` — `findPropsInstanceVariable` requires `left.property.type === Identifier` (a `PrivateIdentifier` never matches)
  - pattern 8: `props-usage-analyzer.ts` — method calls with props are only collected from the constructor body; calls made inside those methods are not followed, and `this.method(props)` is deliberately excluded from the whole-usage fallback
